corehq/apps/zapier/tests.py

```diff
@@ -396,3 +397,122 @@ def test_case_fields(self):
         actual_fields = ZapierCustomFieldCaseResource().obj_get_list(bundle)
         for i in range(len(actual_fields)):
             self.assertEqual(expected_fields[i], actual_fields[i].get_content())
+
+
+class TestZapierCreateCaseAction(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestZapierCreateCaseAction, cls).setUpClass()
+        cls.test_url = "http://commcarehq.org/?domain=fruit&case_type=watermelon&user_id=test_user&user=test"
+        cls.data = {'case_name': 'test1', 'price': '11'}
+        cls.accessor = CaseAccessors('fruit')
+        cls.domain_object = Domain.get_or_create_with_name('fruit', is_active=True)
+        cls.domain = cls.domain_object.name
+        cls.user = WebUser.create(cls.domain, 'test', '******')
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete()
+        cls.domain_object.delete()
+        super(TestZapierCreateCaseAction, cls).tearDownClass()
+
+    def test_create_case(self):
+        factory = RequestFactory()
+        request = factory.post(self.test_url,
+                              data=json.dumps(self.data),
+                              content_type='application/json')
+
+        status = ZapierCreateCase().post(request)
+        self.assertEqual(status.status_code, 200)
+
+        case_id = self.accessor.get_case_ids_in_domain()
+        case = self.accessor.get_case(case_id[0])
+        self.assertEqual('test1', case.get_case_property('name'))
+        self.assertEqual('11', case.get_case_property('price'))
+        self.assertEqual('watermelon', case.get_case_property('type'))
+        self.assertEqual('test_user', case.get_case_property('owner_id'))
+        self.accessor.soft_delete_cases(case_id)
+
+    def test_update_case(self):
+        factory = RequestFactory()
+        request = factory.post(self.test_url,
+                              data=json.dumps(self.data),
+                              content_type='application/json')
+        request.user = self.user
+
+        status = ZapierCreateCase().post(request)
+        self.assertEqual(status.status_code, 200)
+        case_id = self.accessor.get_case_ids_in_domain()
+        case = self.accessor.get_case(case_id[0])
+        self.assertEqual('11', case.get_case_property('price'))
+
+        data = {'case_name': 'test1', 'price': '15', 'case_id': case_id[0]}
+        request = factory.post(self.test_url,
+                               data=json.dumps(data),
+                               content_type='application/json')
+
+        status = ZapierUpdateCase().post(request)
+        self.assertEqual(status.status_code, 200)
+        case = self.accessor.get_case(case_id[0])
+        self.assertEqual('15', case.get_case_property('price'))
+
+        self.accessor.soft_delete_cases(case_id)
```

Comment:
You can follow this [pattern](https://github.com/dimagi/commcare-hq/blob/5e9894d5618b3d3e28fd866868d1e55b909e22f0/docs/forms_and_cases.rst#cleaning-up) instead, which will also delete all the forms, and you won't have to replicate this line in all the tests.